curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-pyramid-generator/660f2b6fd54ac1fc142804dd.md

```diff
@@ -0,0 +1,57 @@
+---
+id: 660f2b6fd54ac1fc142804dd
+title: Step 37
+challengeType: 0
+dashedName: step-37
+---
+
+# --description--
+
+The logic for formatting this pyramid is likely going to get complicated, which means it's a great time to extract that code into a function.
+
+A <dfn>function</dfn> is a block of code that can be reused throughout your application. Functions are declared with the following syntax:
+
+```js
+function name(parameter) {
+
+}
+```
+
+The `function` keyword tells JavaScript that the `name` variable is going to be a function. `parameter` is a variable that represents a value that is passed into the function when it is used. A function may have as many, or as few, <dfn>parameters</dfn> as you'd like. Like a `for` loop, the space between the curly braces is the <dfn>function body</dfn>.
+
+Declare a `padRow` function. Do not create any parameter variables yet. The function body should be empty.
```

Comment:
this is the first time working with camel case.
so that should be explained like it was here

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-9